sep/09-07-2019/README.md

```diff
@@ -0,0 +1,352 @@
+class Solution:
+    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
+```
+
+**Lets try to find a recursive solution to the problem**  
+If we ignore the edge cases we have an almost trivially simple problem.  
+Assume we are given the following linked lists
+
+```
+list1 = 6 -> 2 -> None
+list2 = 1 -> 3 -> None
+
+result = 7 -> 5 -> None
+```
+
+We create a new ListNode with the value being the sum of the two nodes values at the head, then we move onto the next node's value for both lists
+```
+list1 = 6 -> 2
+list2 = 1 -> 3
+
+current values = 6 + 1 = 7
+result = 7
+***
+current values = 2 + 3 = 5
+result = 7 -> 5
+***
+current values = None, None
+result = 7 -> 5 -> None
+```
+
+If we translate that logic into code
+```python
+  def addTwoNumbers(self, l1: ListNode, l2: ListNode, carry = 0) -> ListNode:
+    # If both lists point to None then we should return None since there are no more numbers to add and linked lists should have the tail pointing to None
+    if l1 is None and l2 is None:
+      return None
+    # add the two numbers together and make a new node with that value
+    result_node = ListNode(l1.val + l2.val)
+    # recursive call with the next value of both linked lists
+    result_node.next = self.addTwoNumbers(l1.next, l2.next)
+    return result_node
+```
+
+## Side note about the keyword `is`
+In python we have english word keywords for operators
+```python
+not # !
+or  # ||
+and # &&
+```
+However the keyword `is` is not the same as the equivalency operator `==`  
+Python's `==` is most similar to Ruby's `==`
+```ruby
+# in ruby
+a = [3, 4]
+b = [3, 4]
+a == b # true
+
+c = [4, 3]
+a = c # false
+```
+We get the same results in python
+```python
+# in python
+a = [3, 4]
+b = [3, 4]
+a == b # true
+
+c = [4, 3]
+a = c # false
+```
+
+However if we use `is` then a `is` b returns false
+```python
+# in python
+a = [3, 4]
+b = [3, 4]
+a is b # false
+```
+
+While the two arrays contains the same values in the same order they are not the same array, so if we were to modify one of them then the other array would not be modified
+```python
+a[0] = '!'
+b[1] = '!'
+
+a # ['!', 4]
+b # [3, '!']
+```
+
+In order for `a is b` to be true then we would need to assign a to b
+```python
+a = [3, 4]
+b = a
+
+a is b # true
+```
+
+## Back to the problem, we should start accounting for some edge cases
+The edge cases we should account for is
++ linked lists of different lengths
+```
+list1 = 6 -> 2 -> None
+list2 = 1 -> None
+
+result = 7 -> 2 -> None
+26 + 2 = 27
+```
++ when the values sum is greater or equal 10 then we need to carry
+```
+list1 = 6 -> 2 -> None
+list2 = 5 -> 1 -> None
+
+result = 1 -> 4 -> None
+26 + 15 = 41
+```
++ when the last two values in the linked list sums to a value greater or equal 10 then we need to add an additional node for that carry
+```
+list1 = 6 -> 2 -> None
+list2 = 1 -> 9 -> None
+
+result = 7 -> 1 -> 1 -> None
+26 + 91 = 117
+```
+
+## quick aside for ternaries in python
```

Comment:
Start headings with a capital, since that is the style used in your other headings.